### PR TITLE
[TOR-2762] Added custom stylings for select with full border.

### PR DESCRIPTION
**Pull request by ashraj98**
[TOR-2762](https://torchdental.atlassian.net/browse/TOR-2762)

Separate stylings for select from standard input.
- Had to add a `select--wrapper` class so that `::after` could be used on the wrapper class to add in the chevron icon above the select.
- Added select sample to the component guide.
Screen Shot 2021-03-05 at 3 36 56 PM

### CODE REVIEW

File: src/sass/_forms.scss [hunk 1]
```diff
@@ -11,8 +11,7 @@ label,
 }
 
 input,
-textarea,
-select {
+textarea {
```
Comment (pyen) [inline src/sass/_forms.scss:14]: Not sure removing default styles for the `select` tag is necessarily a good idea. Not all uses are classed with `select--wrapper`. A quick grep of `torchweb/static/app` shows some raw selects, including `BankAccountProfileForm.tsx`.


Can the wrapper styles be rewritten in a way that is compatible with (by overriding) the base `select` styles?

Alternately, ensure all downstream uses require `select--wrapper`, but considering those include `torchweb` as well as `torchdental` repositories, perhaps safer to have a base style for the `<select>` tag.

File: src/sass/utility/mixins/_form-mixins.scss [hunk 2]
```diff
@@ -4,3 +4,8 @@
   border-bottom-color: $color--form-border;
   border-radius: 0;
 }
+
+@mixin inputBorderAll() {
+  border: 1px solid $color--border;
+  border-radius: 4px;
+}
```
Comment (pyen) [inline src/sass/utility/mixins/_form-mixins.scss:11]: This is effectively the same as:
```
  @include border-radius;
  border: $border--light;
```
This pair is in common enough usage that I'd suggest removing the mixin.

**Review by pyen — CHANGES_REQUESTED** [review]

File: src/sass/_forms.scss (hunk 1 above)
Comment (ashraj98) [reply src/sass/_forms.scss:15]: I added it back. I was under the assumption we were updating the styles for all user-facing selects based on the ticket verbiage, so I removed it. I wrote the styles for `.select--wrapper select` before I took out `select` from where you noted, so all the new styles are overriding the old ones. I've added a screenshot as well for reference.

File: src/sass/utility/mixins/_form-mixins.scss (hunk 2 above)
Comment (ashraj98) [reply src/sass/utility/mixins/_form-mixins.scss]: Gotcha, removed this.

**Review by pyen — APPROVED** [review]

**Pull request merged**